src/main/java/com/store/book/service/impl/BookServiceImpl.java

```diff
@@ -14,36 +14,53 @@
 @RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService {
+
     private final BookRepository bookRepository;
+
     private final BookMapper bookMapper;
 
     @Override
-    public BookDto save(CreateBookRequestDto requestDto) {
-        Book book = bookMapper.toModel(requestDto);
-        book.setIsbn("" + (long) (Math.random() * 9_000_000_000_000L) + 1_000_000_000_000L);
-        Book savedBook = bookRepository.save(book);
-        return bookMapper.toDto(savedBook);
+    public BookDto save(CreateBookRequestDto bookDto) {
+
+        Book book = bookMapper.toModel(bookDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
     public List<BookDto> findAll() {
-        return bookRepository.findAll().stream()
+
+        return bookRepository.findAll()
+                .stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<BookDto> findAllByTitle(String title) {
+
+        return bookRepository.findAllByTitleContainingIgnoreCase(title)
+                .stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
 
     @Override
     public BookDto findById(Long id) {
-        Book book = bookRepository.findById(id).orElseThrow(
-                () -> new EntityNotFoundException("Can't find by id " + id)
-        );
+
+        Book book = bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id " + id));
         return bookMapper.toDto(book);
     }
 
     @Override
-    public List<BookDto> getAllByTitle(String title) {
-        return bookRepository.findAllByTitle(title).stream()
-                .map(bookMapper::toDto)
-                .toList();
+    public void deleteById(Long id) {
+
+        bookRepository.findById(id).ifPresentOrElse(
+                bookRepository::delete,
+                () -> {
+                    throw new EntityNotFoundException(
+                            "Can't delete, book with id "
+                                    + id + " not found.");
+                });
```

Comment:
You can call delete only for non-existing records. Use only the delete method and investigate how it works ;)